lib/models/patient/patient.js

```diff
@@ -103,6 +103,29 @@ module.exports = function (gfs) {
       });
     }
 
+    // If we're updating a user's default patient info, we also update the user's info
+    // Only for firstName, lastName, phone
+    PatientSchema.post("save", function (next) {
+        if (!this.isNew && this.me) {
+            mongoose.model("User").findOne({ email: this.creator }).exec().then((user) => {
+                if (user !== null) {
+                    if (user.firstName !== this.firstName ||
+                        user.lastName !== this.lastName ||
+                        user.phone !== this.phone) {
+                        user.firstName = this.firstName;
+                        user.lastName = this.lastName;
+                        user.phone = this.phone;
+                        return user.save();
+                    }
+                }
+                // if no user, we just continue
+                return Promise.resolve();
+            }).then(() => next()).catch(err => next(err));
```

Comment:
The following was run entirely locally.

When I test this using the new form in orange-web PR https://github.com/amida-tech/orange-web/pull/63 my Orange API errors here on line 123 of this file. Below is the error (log level debug). It says "next" is undefined.
```
{"level":"info","message":"HTTP PUT /v1/patients/5 200 92ms","meta":{"res":{"statusCode":200,"body":{"access":"none","group":null,"access_prime":"write","access_family":"write","access_anyone":"write","first_name":"Aaron","last_name":"Houlihan P0","meditations":{"count":0,"sum":0},"birthdate":null,"sex":"unspecified","phone":"1112223332","creator":"aaron+p0@amida.com","me":true,"instructorEmail":null,"sharesCount":28,"avatar":"/v1/patients/5/avatar.png","success":true,"id":5}},"req":{"url":"/v1/patients/5","headers":{"host":"localhost:5000","connection":"keep-alive","content-length":"72","x-client-secret":"testsecret","accept":"application/json","origin":"http://localhost:1776","authorization":"Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NCwidXVpZCI6ImNkZjMzNjk3LTFlZDQtNDViOS04ZDBmLTNmMGRlMjBiZjkzMiIsInVzZXJuYW1lIjoiYWFyb24rcGEwQGFtaWRhLmNvbSIsImVtYWlsIjoiYWFyb24rcGEwQGFtaWRhLmNvbSIsInNjb3BlcyI6WyJwcm9ncmFtQWRtaW5pc3RyYXRvciJdLCJpYXQiOjE1NDc2NzQwODgsImV4cCI6MTU1MDI2NjA4OH0._rMYLMWE043W7CJ-dOQCNSOA_5HUDZCe6HWb0_uGQO0","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36","content-type":"application/json","referer":"http://localhost:1776/accounts","accept-encoding":"gzip, deflate, br","accept-language":"en-US,en;q=0.9"},"method":"PUT","httpVersion":"1.1","originalUrl":"/v1/patients/5","query":{},"body":{"firstName":"Aaron P02","lastName":"Houlihan P02","phone":"1112223332"}},"responseTime":92}}
(node:10209) UnhandledPromiseRejectionWarning: TypeError: next is not a function
    at mongoose.model.findOne.exec.then.then.catch.err (/Users/houli/docs/gitrepos/amida-tech/orange-api/lib/models/patient/patient.js:123:48)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:10209) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
(node:10209) UnhandledPromiseRejectionWarning: TypeError: next is not a function
    at mongoose.model.findOne.exec.then.then.catch.err (/Users/houli/docs/gitrepos/amida-tech/orange-api/lib/models/patient/patient.js:123:48)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:10209) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 5)
(node:10209) UnhandledPromiseRejectionWarning: TypeError: next is not a function
    at mongoose.model.findOne.exec.then.then.catch.err (/Users/houli/docs/gitrepos/amida-tech/orange-api/lib/models/patient/patient.js:123:48)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:10209) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)
```